jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/JaxRsResourceBase.java

```diff
@@ -475,4 +477,32 @@ protected void verifyNumberOfElements(int actual, int expected, String errorMess
         Preconditions.checkArgument(actual == expected, errorMessage);
     }
 
+    protected Response createPaymentResponse(UriInfo uriInfo, Payment payment, TransactionType transactionType, @Nullable String transactionExternalKey) {
+        final PaymentTransaction createdTransaction = findCreatedTransaction(payment, transactionType, transactionExternalKey);
+        switch (createdTransaction.getTransactionStatus()) {
+            case PENDING:
+            case SUCCESS:
+                return uriBuilder.buildResponse(uriInfo, PaymentResource.class, "getPayment", payment.getId());
+            case PAYMENT_FAILURE:
+                return Response.status(402).build();
+            case PAYMENT_SYSTEM_OFF:
+            case UNKNOWN:
+                return Response.status(Status.SERVICE_UNAVAILABLE).build();
```

Comment:
UKNOWN could mean other things, not necessarily a timeout. I think we should keep it as a 503 so we can differentiate from timeouts. And then, we can have the plugins throwing the right exception in case of a timeout, so it's handled by the exception mapper.